src/Explorer/Panes/SaveQueryPanel/index.tsx

```diff
@@ -0,0 +1,174 @@
+  const genericPaneProps: GenericRightPaneProps = {
+    container: explorer,
+    formError: formError,
+    formErrorDetail: formErrorsDetails,
+    id: "saveQueryPane",
+    isExecuting: isLoading,
+    title,
+    submitButtonText: canSaveQueries() ? "Save" : "Complete setup",
+    onClose: () => closePanel(),
+    onSubmit: () => {
+      canSaveQueries() ? submit() : setupQueries();
+    },
+  };
+
+  const submit = (): void => {
+    setFormError("");
+    setFormErrorsDetails("");
+    if (!canSaveQueries()) {
+      setFormError("Cannot save query");
+      setFormErrorsDetails("Failed to save query: account not set up to save queries");
+      logConsoleError("Failed to save query: account not setup to save queries");
+    }
+
+    const queryTab = explorer && (explorer.tabsManager.activeTab() as QueryTab);
+    const query: string = queryTab && queryTab.sqlQueryEditorContent();
+    if (!queryName || queryName.length === 0) {
+      setFormError("No query name specified");
+      setFormErrorsDetails("No query name specified. Please specify a query name.");
+      logConsoleError("Could not save query -- No query name specified. Please specify a query name.");
+      return;
+    } else if (!query || query.length === 0) {
+      setFormError("Invalid query content specified");
+      setFormErrorsDetails("Invalid query content specified. Please enter query content.");
+      logConsoleError("Could not save query -- Invalid query content specified. Please enter query content.");
+      return;
+    }
+
+    const queryParam: Query = {
+      id: queryName,
+      resourceId: explorer.queriesClient.getResourceId(),
+      queryName: queryName,
+      query: query,
+    };
+    const startKey: number = traceStart(Action.SaveQuery, {
+      dataExplorerArea: Areas.ContextualPane,
+      paneTitle: title,
+    });
+    setLoadingTrue();
+    explorer.queriesClient.saveQuery(queryParam).then(
+      () => {
+        setLoadingFalse();
+        queryTab.tabTitle(queryParam.queryName);
+        queryTab.tabPath(`${queryTab.collection.databaseId}>${queryTab.collection.id()}>${queryParam.queryName}`);
+        traceSuccess(
+          Action.SaveQuery,
+          {
+            dataExplorerArea: Areas.ContextualPane,
+            paneTitle: title,
+          },
+          startKey
+        );
+        closePanel();
+      },
+      (error: Error) => {
+        setLoadingFalse();
+        const errorMessage = getErrorMessage(error);
+        setFormError("Failed to save query");
+        setFormErrorsDetails(`Failed to save query: ${errorMessage}`);
+        traceFailure(
+          Action.SaveQuery,
+          {
+            dataExplorerArea: Areas.ContextualPane,
+            paneTitle: title,
+            error: errorMessage,
+            errorStack: getErrorStack(error),
+          },
+          startKey
+        );
+      }
+    );
+  };
+
+  const setupQueries = async (): Promise<void> => {
+    if (!explorer) {
+      return;
```

Comment:
This probably should never happen.
I would remove it, so it actually breaks, rather than fail silently.

Reply:
I removed